### PR TITLE
framework/eventloop: Fix issues about loop closing

### PR DESCRIPTION
**Pull request by jeongarmy**
- framework/eventloop: add closing loop when all works in loop are finished
- external/libtuv: update missing logics related to signal and loop
- framework/eventloop: fix loop closing procedure

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [313d696659fb6396e1ed8e3c8e835b51be45a114] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [313d696659fb6396e1ed8e3c8e835b51be45a114] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@jeongarmy I am sorry but I can't understand what you did. Could you explain this PR?

**Comment by seinfra** [comment]
Target : [9904fac38bbfc2d168b3b47a8c3163e387bf7b28] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9904fac38bbfc2d168b3b47a8c3163e387bf7b28] - Code Rule Check OK.

File: framework/src/eventloop/eventloop_loop.c [hunk 1]
```diff
@@ -83,25 +83,24 @@ int eventloop_loop_run(void)
...
 }
 
 int eventloop_loop_stop(void)
 {
-	el_loop_t *loop;
+	int index;
 
-	loop = get_app_loop();
-	if (loop == NULL) {
+	index = PID_HASH(getpid());
```
Comment (sunghan-chang) [inline framework/src/eventloop/eventloop_loop.c]: How about use another MACRO instead of ```PID_HASH(getpid())```?
like get_this_thread_loop_idx?

**Comment by seinfra** [comment]
Target : [5261b45c42ce5db5d3fd9f6739cd50a596993d46] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5261b45c42ce5db5d3fd9f6739cd50a596993d46] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7d6f6474d816b111b90f7f4e9e4302d93530b08c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7d6f6474d816b111b90f7f4e9e4302d93530b08c] - Code Rule Check OK.

File: framework/src/eventloop/eventloop_loop.c (hunk 1 above)
Comment (jeongarmy) [reply framework/src/eventloop/eventloop_loop.c]: Yes, I agree with your opinion. I will update all later.

**Comment by seinfra** [comment]
Target : [b5a2af4c206364daf1f638612a01e5b6260384a5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b5a2af4c206364daf1f638612a01e5b6260384a5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0499ac07023fb5a3ae5cf181d57f677890ef48f8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0499ac07023fb5a3ae5cf181d57f677890ef48f8] - Code Rule Check OK.

**Pull request merged**